hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -727,6 +734,13 @@ public void setupWriteClient() throws IOException {
 
   private void reInitWriteClient(Schema sourceSchema, Schema targetSchema) throws IOException {
     LOG.info("Setting up new Hoodie Write Client");
+    final boolean isDropPartitionColumns = props.getBoolean(DROP_PARTITION_COLUMNS.key());
+    if (isDropPartitionColumns) {
+      String partitionColumns = HoodieSparkUtils.getPartitionColumns(keyGenerator, props);
```

Comment:
we already get these once at L 480 ish. is it possible to re-use them.

Reply:
updated it.